src/compiler/transformers/ts.ts

```diff
@@ -2114,10 +2115,14 @@ namespace ts {
          * This function will be called when one of the following conditions are met:
          * - The node has an accessibility modifier.
          * - The node has a questionToken.
+         * - The node's text is "this".
          *
          * @param node The parameter declaration node.
          */
         function visitParameter(node: ParameterDeclaration) {
+            if ((node.name as Identifier).text === "this") {

```

Comment:
Minor: You can check by `if ((node.name as Identifier).originalKeywordKind=== SyntaxKind.ThisKeyword)`